### PR TITLE
on anchor clicked contract change

### PR DESCRIPTION
**Pull request by dptejas**
Currently, adaptiveCard instance is passed to handle anchor click. Change here is to pass text block element from which adaptiveCard instance can be derived if required.

### CODE REVIEW

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -5014,11 +5014,11 @@ function raiseImageLoadedEvent(image: Image) {
     }
 }
 
-function raiseAnchorClickedEvent(element: CardElement, anchor: HTMLAnchorElement): boolean {
-    let card = element.getRootElement() as AdaptiveCard;
+function raiseAnchorClickedEvent(textBlock: TextBlock, anchor: HTMLAnchorElement): boolean {
+    let card = textBlock.getRootElement() as AdaptiveCard;
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: There is no need to change this. Please keep it as CardElement. We may support inline actions in elements that are not TextBlock in the future.

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 2]
```diff
@@ -5417,7 +5417,7 @@ export class AdaptiveCard extends ContainerWithActions {
         return true;
     }
 
-    onAnchorClicked: (rootCard: AdaptiveCard, anchor: HTMLAnchorElement) => boolean = null;
+    onAnchorClicked: (textBlock: TextBlock, anchor: HTMLAnchorElement) => boolean = null;
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Should be `element: CardElement`

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 3]
```diff
@@ -5325,7 +5325,7 @@ export class AdaptiveCard extends ContainerWithActions {
     static readonly elementTypeRegistry = new ElementTypeRegistry();
     static readonly actionTypeRegistry = new ActionTypeRegistry();
 
-    static onAnchorClicked: (rootCard: AdaptiveCard, anchor: HTMLAnchorElement) => boolean = null;
+    static onAnchorClicked: (textBlock: TextBlock, anchor: HTMLAnchorElement) => boolean = null;
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: Should be `element: CardElement`

**Comment by dclaux** [comment]
Please also update the visualizer application which uses this feature.

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 4]
```diff
@@ -5325,7 +5325,7 @@ export class AdaptiveCard extends ContainerWithActions {
     static readonly elementTypeRegistry = new ElementTypeRegistry();
     static readonly actionTypeRegistry = new ActionTypeRegistry();
 
-    static onAnchorClicked: (textBlock: TextBlock, anchor: HTMLAnchorElement) => boolean = null;
+    static onAnchorClicked: (elementd: CardElement, anchor: HTMLAnchorElement) => boolean = null;
```
Comment (dclaux) [inline source/nodejs/adaptivecards/src/card-elements.ts]: `elementd` --> `element`

**Comment by dptejas** [comment]
> Please also update the visualizer application which uses this feature.

Realized that changes are required in designer as well. Made the appropriate changes

File: source/nodejs/adaptivecards-designer/src/containers/outlook-container.ts [hunk 5]
```diff
@@ -116,9 +116,9 @@ export class OutlookContainer extends HostContainer {
         }
     }
 
-    public anchorClicked(rootCard: Adaptive.AdaptiveCard, anchor: HTMLAnchorElement): boolean {
+    public anchorClicked(element: Adaptive.CardElement, anchor: HTMLAnchorElement): boolean {
         var regEx = /^action:([a-z0-9]+)$/ig;
-
+        var rootCard = element.getRootElement();
```
Comment (dclaux) [inline source/nodejs/adaptivecards-designer/src/containers/outlook-container.ts]: We should check that this is an AdaptiveCard. getRootElement returns a CardElement.

File: source/nodejs/adaptivecards-visualizer/src/containers/outlook.ts [hunk 6]
```diff
@@ -120,8 +120,9 @@ export class OutlookContainer extends HostContainer {
         }
     }
 
-    public anchorClicked(rootCard: Adaptive.AdaptiveCard, anchor: HTMLAnchorElement): boolean {
+    public anchorClicked(element: Adaptive.CardElement, anchor: HTMLAnchorElement): boolean {
         var regEx = /^action:([a-z0-9]+)$/ig;
+        var rootCard = element.getRootElement();
```
Comment (dclaux) [inline source/nodejs/adaptivecards-visualizer/src/containers/outlook.ts]: Should check that this is actually an AdaptiveCard instance.

Comment (dptejas) [reply source/nodejs/adaptivecards-visualizer/src/containers/outlook.ts]: making var rootCard = element.getRootElement() as AdaptiveCard works?

Comment (dclaux) [reply source/nodejs/adaptivecards-visualizer/src/containers/outlook.ts]: Yes that should be enough given there is a `if (rootCard)` statement after.

**Pull request merged**